### PR TITLE
Principle point

### PR DESCRIPTION
**Pull request by Mickmcginnis**

### CODE REVIEW

**Comment by hazel-sudz** [comment]
closes #15

File: dataset.py [hunk 1]
```diff
@@ -362,11 +362,21 @@ def _get_single_item(self, idx, image_height):
...
 
         # Load intrinsics.
-        focal_length = float(np.loadtxt(self.calibration_files[idx]))
+        k = np.loadtxt(self.calibration_files[idx])
+        if k.size == 1:
+            focal_length = float(k)
+        else:
+            k = k.tolist()
+            focal_length = [k[0][0], k[1][1]]
+            centre_point = [k[0][2], k[1][2]]
```
Comment (hazel-sudz) [inline dataset.py]: @tcavallari we can change this to just be 4 values or whatever you think is more appropriate. Currently we assume that that it's a 3x3 intrinsics matrix like so: 

```
fx 0 cx
0 fy cy
0 0 1
```

just for the sake of being consistent with literature notation

Comment (tcavallari) [reply dataset.py]: 3x3 matrix is fine. Can you add a check to make sure that's the actual shape of the input matrix?

File: dataset.py [hunk 2]
```diff
@@ -362,11 +362,21 @@ def _get_single_item(self, idx, image_height):
...
 
         # The image will be scaled to image_height, adjust focal length as well.
         f_scale_factor = image_height / image.shape[0]
-        focal_length *= f_scale_factor
+        if centre_point:
```
Comment (tcavallari) [inline dataset.py:380]: This results in a `NameError` if `centre_point` is not defined (which happens when we read a single focal length). Can you initialize it to `None` in the if/else above?

File: dataset.py [hunk 3]
```diff
@@ -484,11 +494,18 @@ def _get_single_item(self, idx, image_height):
...
         # Create the intrinsics matrix.
         intrinsics = torch.eye(3)
-        intrinsics[0, 0] = focal_length
-        intrinsics[1, 1] = focal_length
-        # Hardcode the principal point to the centre of the image.
-        intrinsics[0, 2] = image.shape[2] / 2
-        intrinsics[1, 2] = image.shape[1] / 2
+        
+        # Hardcode the principal point to the centre of the image unless otherwise specified.
+        if centre_point:
```
Comment (tcavallari) [inline dataset.py:504]: Same, this also results in a `NameError`.

**Review by tcavallari — CHANGES_REQUESTED** [review]

**Comment by Mickmcginnis** [comment]
Made both changes and tested. Thanks!

File: dataset.py (hunk 1 above)
Comment (tcavallari) [reply dataset.py]: Thanks!

**Review by tcavallari — APPROVED** [review]

**Pull request merged**